src/Grid.php

```diff
@@ -129,9 +135,42 @@ public function addAction($label, $action)
         $this->actions->addAction($label, $action);
     }
 
+    /**
+     * Apply ordering to the current model as per the sort parameters.
+     */
+    public function applySort()
+    {
+        $sortby = $this->app->stickyGET($this->name.'_sort', null);
+        $desc = false;
+        if ($sortby && $sortby[0] == '-') {
+            $desc = true;
+            $sortby = substr($sortby, 1);
+        }
+
+        $this->table->sortable = true;
+
+        if ($sortby && isset($this->table->columns[$sortby]) && $this->model->hasElement($sortby) instanceof \atk4\data\Field) {
+            $this->model->setOrder($sortby, $desc);
+            $this->table->sort_by = $sortby;
+            $this->table->sort_order = $desc ? 'descending' : 'ascending';
+        }
+
+        $this->table->on('click', 'thead>tr>th', new jsReload($this, [$this->name.'_sort'=>(new jQuery())->data('column')]));
+    }
+
     public function setModel(\atk4\data\Model $model, $columns = null)
     {
-        return $this->model = $this->table->setModel($model, $columns);
+        $this->model = $this->table->setModel($model, $columns);
+
+        if ($this->sortable === null) {
```

Comment:
we will try and detect capability of a model to set order. E.g. if Array persistence does not support, then this will be disabled. Similar with pagination.